salt/utils/event.py

```diff
@@ -735,39 +735,65 @@ def destroy(self):
         if self._run_io_loop_sync and not self.keep_loop:
             self.io_loop.close()
 
+    def _fire_ret_load_specific_fun(self, load, fun):
+        '''
+        Helper function for fire_ret_load
+        '''
+        if isinstance(load['fun'], list):
+            # Multi-function job
+            ret = load.get('return', {})
+            ret = ret.get(fun, {})
+            # This was already validated to exist and be non-zero in the
+            # caller.
+            retcode = load['retcode'][fun]
+        else:
+            # Single-function job
+            ret = load.get('return', {})
+            retcode = load['retcode']
+
+        try:
+            for tag, data in six.iteritems(ret):
+                data['retcode'] = retcode
+                tags = tag.split('_|-')
+                if data.get('result') is False:
+                    self.fire_event(
+                        data,
+                        '{0}.{1}'.format(tags[0], tags[-1])
+                    )  # old dup event
+                    data['jid'] = load['jid']
+                    data['id'] = load['id']
+                    data['success'] = False
+                    data['return'] = 'Error: {0}.{1}'.format(
+                        tags[0], tags[-1])
+                    data['fun'] = fun
+                    data['user'] = load['user']
+                    self.fire_event(
+                        data,
+                        tagify([load['jid'],
+                                'sub',
+                                load['id'],
+                                'error',
+                                fun],
+                               'job'))
+        except Exception:
```

Comment:
Bare excepts like this have bitten us in the past. Is this designed to catch unknown exceptions or is there something that you anticipate hitting this? We might want to log here, if appropriate.